frontend/packages/topology/src/operators/components/OperatorBackedService.tsx

```diff
@@ -37,30 +40,52 @@ export const obsDropTargetSpec = (
   any,
   { canDrop: boolean; dropTarget: boolean; edgeDragging: boolean },
   NodeComponentProps
-> => ({
-  accept: [CREATE_CONNECTOR_DROP_TYPE],
-  canDrop: (item, monitor, props) => {
-    if (!serviceBinding) {
-      return false;
-    }
+> => {
+  const getSourceNode = (monitor) =>
+    monitor.getItem() instanceof OdcBaseEdge ? monitor.getItem().getSource() : monitor.getItem();
 
-    if (isEdge(item)) {
-      return canDropEdgeOnNode(monitor.getOperation()?.type, item, props.element);
-    }
-    if (!props.element || item === props.element) {
-      return false;
-    }
-    return !props.element.getTargetEdges().find((e) => e.getSource() === item);
-  },
-  collect: (monitor, props) => {
-    return {
-      canDrop: serviceBinding && highlightNode(monitor, props.element),
-      dropTarget: monitor.isOver({ shallow: true }),
-      edgeDragging: nodesEdgeIsDragging(monitor, props),
-    };
-  },
-  dropHint: 'createServiceBinding',
-});
+  return {
+    accept: [CREATE_CONNECTOR_DROP_TYPE],
+    canDrop: (item, monitor, props) => {
+      if (!serviceBinding) {
+        return false;
+      }
+
+      const sourceNode = getSourceNode(monitor);
+
+      if (
+        sourceNode instanceof OdcBaseNode &&
+        sourceNode.getData().resource.kind !== DeploymentModel.kind &&
+        sourceNode.getData().resource.kind !== DeploymentConfigModel.kind &&
+        sourceNode.getData().resource.kind !== ServiceModel.kind
+      ) {
+        return false;
+      }
+      if (isEdge(item)) {
+        return canDropEdgeOnNode(monitor.getOperation()?.type, item, props.element);
+      }
+      if (!props.element || item === props.element) {
+        return false;
+      }
+      return !props.element.getTargetEdges().find((e) => e.getSource() === item);
+    },
+    collect: (monitor, props) => {
+      const sourceNode = getSourceNode(monitor);
+      return {
+        canDrop:
+          serviceBinding &&
+          sourceNode instanceof OdcBaseNode &&
+          (sourceNode.getData().resource.kind === DeploymentModel.kind ||
+            sourceNode.getData().resource.kind === DeploymentConfigModel.kind ||
+            sourceNode.getData().resource.kind === ServiceModel.kind) &&
+          highlightNode(monitor, props.element),
+        dropTarget: monitor.isOver({ shallow: true }),
+        edgeDragging: nodesEdgeIsDragging(monitor, props),
+      };
+    },
+    dropHint: 'createServiceBinding',
+  };
+};
```

Comment:
@debsmita1 We show other workloads such as DaemonSets etc. With this binding with such resources wont be possible. We should first make sure what is supported and what is not. Then if only a pods are not supported then we should just add a check for that only and allow binding with all other resources.

Reply:
as of today, we show the `Create Service Binding` action only for Deployment, DeploymentConfig, and Knative Service. So I kept it consistent for the drag-and-drop action as well

Reply:
Pods, Jobs and CronJobs are not supported. I have updated the PR accordingly. PTAL